### PR TITLE
Check the slide_view constructor precondition added by LWG-3711

### PR DESCRIPTION
**Pull request by CaseyCarter**
Towards #2872.

This is "blocked on WP" because I add a citation for a paragraph that doesn't yet exist in the working draft.

### CODE REVIEW

File: tests/std/tests/P2442R1_views_slide_death/test.cpp [hunk 1]
```diff
@@ -0,0 +1,34 @@
+// Copyright (c) Microsoft Corporation.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#define _CONTAINER_DEBUG_LEVEL 1
+
+#include <cassert>
+#include <ranges>
+#include <span>
```
Comment (CaseyCarter) [inline tests/std/tests/P2442R1_views_slide_death/test.cpp:8]: This no longer uses `<span>`

File: stl/inc/ranges [hunk 2]
```diff
@@ -6058,7 +6058,11 @@ namespace ranges {
 
         constexpr explicit slide_view(_Vw _Range_, const range_difference_t<_Vw> _Count_) noexcept(
             is_nothrow_move_constructible_v<_Vw>) /* strengthened */
-            : _Range(_STD move(_Range_)), _Count{_Count_} {}
+            : _Range(_STD move(_Range_)), _Count{_Count_} {
+#if _CONTAINER_DEBUG_LEVEL > 0
+            _STL_VERIFY(_Count > 0, "The window size must be positive (NFIXME [range.slide.view]/1)");
```
Comment (CaseyCarter) [inline stl/inc/ranges:6063]: This `FIXME` must be corrected before checkin.

File: tests/std/tests/P2442R1_views_slide_death/test.cpp [hunk 3]
```diff
@@ -0,0 +1,34 @@
+// Copyright (c) Microsoft Corporation.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#define _CONTAINER_DEBUG_LEVEL 1
+
+#include <cassert>
```
Comment (StephanTLavavej) [inline tests/std/tests/P2442R1_views_slide_death/test.cpp:6]: This test isn't using `assert()` directly.

**Review by StephanTLavavej — COMMENTED** [review]
Looks good modulo nitpick.

File: stl/inc/ranges (hunk 2 above)
Comment (StephanTLavavej) [reply stl/inc/ranges:6063]: https://isocpp.org/files/papers/N4917.pdf is now available.

**Review by StephanTLavavej — APPROVED** [review]

**Comment by StephanTLavavej** [comment]
I've pushed a conflict-free merge with `main`, followed by the new WP number. Additionally, I noticed that this was using a strict matrix but would pass with non-strict, so I changed that too (as #3015 was created after this PR).

**Comment by StephanTLavavej** [comment]
I'm mirroring this to the MSVC-internal repo - please notify me if any further changes are pushed.

**Comment by StephanTLavavej** [comment]
Thanks for taking care of this LWG issue resolution! :white_check_mark: :tada: :smile_cat:

**Pull request merged**